File: Practica 3/Java/Estructura_De_datos/Pila.java
package Estructura_De_datos;

public class Pila {
    // atributos
    private long[] arreglo;
    private int top;
    private int n;

    // metodos
    public Pila(int n){
        this.arreglo = new long[n];
        this.top = -1;
        this.n = n;
    }
    public void push(int e){
        if (isFull()){
            System.out.println("Pila llena");
        }
        else{
            top++;
            arreglo[top] = e;
        }
    }
    public long pop(){
        if (isEmpty()){
            System.out.println("Pila vacia");
            return -1;
        }
        else{
            long e = arreglo[top];
            top--;
            return (int) e;
        }
    }
    public long peek(){
        if (isEmpty()){
            System.out.println("Pila vacia");
        }
        return (int)arreglo[top];
    }

    public boolean isEmpty() {
        return top == -1;
    }

    public boolean isFull() {
        return top == n - 1;
    }

    public static void main(String[] args){
        Pila pila = new Pila(10);
        pila.push(10);
        pila.push(20);
        pila.push(30);
        System.out.println(pila.peek());
        System.out.println("pop() = "+pila.pop());
        System.out.println(+pila.peek());
        System.out.println("pop() = "+pila.pop());
        System.out.println("pop() = "+pila.pop());
    }
}

File: Practica 3/Python/Pila.py
class Pila:
    def __init__(self, n):
        self.arreglo = [None] * n
        self.top = -1
        self.n = n

    def push(self, e):
        if self.isFull():
            print("Pila llena")
        self.top += 1
        self.arreglo[self.top] = e

    def pop(self):
        if self.isEmpty():
            print("Pila vacía")
        elemento = self.arreglo[self.top]
        self.top -= 1
        return elemento

    def peek(self):
        if self.isEmpty():
            print("Pila vacía")
        return self.arreglo[self.top]

    def isEmpty(self):
        return self.top == -1

    def isFull(self):
        return self.top == self.n - 1

pila = Pila(10)  # Crea una pila de tamaño 10
pila.push(10)
pila.push(20)
pila.push(30)
print(pila.peek())  # Muestra el elemento en el tope (30)
print("pop() =", pila.pop())  # Elimina y muestra el tope (30)
print(pila.peek())  # Muestra el nuevo tope (20)
print("pop() =", pila.pop())  # Elimina y muestra el tope (20)
print("pop() =", pila.pop())  # Elimina y muestra el tope (10)